game/game.py

```diff
@@ -440,6 +434,36 @@ def initialize_turn(self) -> None:
 
         self.plan_procurement()
 
+    def plan_missions(self, is_player: bool, is_replan: bool = False):
+
+        if is_player:
+            coalition = "Blue"
+
+        else:
+            coalition = "Red"
+
+        if is_replan:
+            # Reset ATO, Airwing and Inventory that  flights can be replanned
+            self.ato_for(is_player).clear()
+            if is_player:
+                self.blue_air_wing.reset()
+                self.blue_procurement_requests.clear()
+            else:
+                self.red_air_wing.reset()
+                self.red_procurement_requests.clear()
+
+            self.aircraft_inventory.reset()
+            for cp in self.theater.controlpoints:
+                self.aircraft_inventory.set_from_control_point(cp)
```

Comment:
Looking at `initialize_turn` I think there are a few more things this is missing. It's not just the procurement _requests_ that need to be cleared; the existing order need to be cleared and reprocessed as well. Threat zones need to be recreated. Transports need to be replanned (because annoyingly that's not a part of regular ATO planning).

Rather than splitting this part out, I think the safer (and simpler) path will be to call `initialize_turn` when SAMs are bought and sold. You'll want to add a `red_only: bool = False` or whatever so that it doesn't reset the player's actions presumably.

Reply:
I went through it, debugged any function and put everything related to replanning in the `initialize_turn` as proposed. Should be working good right now, even on cheating (first time i ever used this).
May i have overseen something? Otherwise im ready and already rebased it